packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -233,7 +233,44 @@ async function fetchLatestSnapshotCore(
                     snapshotOptions,
                     controller,
                 );
-                const snapshot = response.odspSnapshotResponse.content;
+
+                const odspResponse = response.odspResponse;
+                const contentType = odspResponse.headers.get("content-type");
+                odspResponse.propsToLog = {
+                    ...odspResponse.propsToLog,
+                    contentType,
+                    accept: response.requestHeaders.accept,
+                };
+                let parsedSnapshotContents: IOdspResponse<ISnapshotContents> | undefined;
+                try {
+                    if (contentType === "application/json") {
+                        const text = await odspResponse.content.text();
+                        odspResponse.propsToLog.bodySize = text.length;
+                        const content: IOdspSnapshot = JSON.parse(text);
+                        validateBlobsAndTrees(content);
+                        const snapshotContents: ISnapshotContents = convertOdspSnapshotToSnapshotTreeAndBlobs(content);
+                        parsedSnapshotContents = { ...odspResponse, content: snapshotContents };
+                    } else {
+                        assert(contentType === "application/ms-fluid", 0x2c3 /* "Content type should be application/ms-fluid" */);
```

Comment:
It feels like some of these asserts should be checks (IF statements) with proper telemetry. I.e. if we hit this condition, it would be great to throw an error object with contentType on it as a property.

Reply:
This change will do that. I am catching the error and adding the appropriate props. I talked to william and he said that the content type could only be either json or ms-fluid. So not removing assert for now, but adding the apt props on error so that we can debug later on and ask the server people.